proposals/291-loeschkonzept-v2.md

```diff
@@ -0,0 +1,369 @@
+# Löschkonzept v2
+
+Das aktuelle Konzept zum Löschen von Räumen und Inhalten in [TI-M Basis] wird in
+Kommentierungsworkshops und Supportanfragen regelmäßig hinterfragt und
+kritisiert. Dabei fallen insbesondere die folgenden Probleme auf:
+
+- Das Konzept ist fragmentiert. Anforderungen tauchen in verschiedenen Kapiteln
+  und ohne übergreifende Erläuterung auf. Das Konzept und dessen Motivation
+  holistisch zu verstehen ist daher sehr schwierig.
+- Die einzelnen Anforderungen sind teilweise überlappend oder lassen zu viel
+  Spielraum für Interpretationen. Beispielsweise sind etwaige Wechselwirkung
+  zwischen client- und serverseitigem Löschen oder die Auswirkungen des Löschens
+  auf die Föderation nicht klar benannt. Unterschiedliche Interpretationen
+  können zu unterschiedlichen Implementierungen und im schlimmsten Fall zu
+  Inkompatibilitäten führen.
+- Die Anforderungen entsprechen teilweise nicht den Bedürfnissen und Erwartungen
+  von Nutzern, Organisationen und Anbietern. Rückmeldungen aus der
+  TI-Modellregion Hamburg z. B. haben ergeben, dass Nutzer nicht verstehen warum
+  sie regelmäßig gefragt werden ob sie Räume, die älter als 6 Monate sind
+  löschen möchten und ob diese Löschung Auswirkungen auf andere Raumteilnehmer
+  hat.
+
+Im hier vorliegenden Proposal wird ausgehend von User Stories ein Vorschlag für
+ein verständliches, sinnvolles und begründetes Löschkonzept präsentiert, das
+obige Probleme löst.
+
+## Änderungsvorschlag
+
+Zur Erarbeitung eines Änderungsvorschlags wurden zunächst die Bedürfnisse von
+Nutzern, Organisationen und Anbietern durch gezielte Nachfragen bei
+Industriepartnern zusammengetragen. Aus diesen Bedürfnissen wurden im nächsten
+Schritt User Stories abgeleitet und gruppiert. Im Folgenden werden die
+gruppierten User Stories und die daraus abgeleiteten Änderungsvorschläge
+vorgestellt.
+
+Ausgangspunkt der Vorschläge ist, dass alle aktuellen Anforderung zum Löschen
+entfallen. Diese Anforderungen sind der Vollständigkeit halber im nachfolgenden
+Abschnitt "Aktuelle Löschanforderungen" aufgelistet. Des Weiteren sind die neuen
+Anforderungen als grobe Vorschläge zu verstehen, die nicht zwingend mit
+identischem Wortlaut in die Spezifikation übernommen werden sollen.
+
+### Serverseitiges Löschen
+
+**Story 1**
+
+- Als Anbieter
+- möchte ich nicht mehr benötigte Räume und deren Inhalte löschen
+- damit ich die Speicherkosten meines Homeservers reduzieren kann.
```

Comment:
Mit in die Betrachtung eingebracht werden sollte, dass Nachrichten nicht wirklich viel Speicher verbrauchen. Aus Sicht des Betreibers ist das mMn nicht ausschlaggebend, wenn es um Kosten geht.

Anders sieht es mit Medien aus. Jedoch weden genau diese (bei strenger Befolgung der Matrix Spec) auch mit den vorgeschlagenen Änderungen nicht gelöscht, wenn Nachrichten oder Räume entfernt werden. Daher denke ich, dass man nach aktuellen Stand der Matrix-Spezifikation die Stories mit bezug zu "Speicherkosten" für nicht-relevant einstufen könnte.

Reply:
Die Medien stecken hier bei "Inhalten" implizit mit drin und werden bei der Löschung eigentlich auch mit [benannt](https://github.com/gematik/api-ti-messenger/pull/291/files#diff-e92774fae81e7e1998d8643240e308a42744cb1bbc27e5064b57b399ce597450R84-R94).

Reply:
sorry, übersehen. Dann kann das hier zugemacht werden.